dist/src/test/java/io/camunda/application/commons/logging/StackdriverLayoutTest.java

```diff
@@ -0,0 +1,229 @@
+
+    // when
+    logger.atError().withLocation(DUMMY_LOCATION).withThrowable(exception).log(LOG_MESSAGE);
+
+    // then
+    assertMessageMatchesWithException(exception);
+  }
+
+  @ParameterizedTest
+  @MethodSource("severityProvider")
+  void shouldMapSeverity(final String expectedSeverity, final Level actualLevel) {
+    // given
+
+    // when
+    logger.atLevel(actualLevel).withLocation(DUMMY_LOCATION).log(LOG_MESSAGE);
+
+    // then
+    assertMessageMatches(expectedSeverity);
+  }
+
+  @SuppressWarnings("unused")
+  @Test
+  void shouldLogContextMap() {
+    // given
+
+    // when
+    try (final var bar = MDC.putCloseable("foo", "bar");
+        final var baz = MDC.putCloseable("baz", "qux")) {
+      logger.atDebug().withLocation(DUMMY_LOCATION).log(LOG_MESSAGE);
+    }
+
+    // then - note that Log4j2 will print out the context map with keys alphabetically sorted
+    assertMessageMatchesWithContext(
+        "DEBUG",
+        """
+        "contextMap":{"baz":"qux","foo":"bar"}""");
+  }
+
+  @Test
+  void shouldLog() {
+    // given
+
+    // when
+    logger.atDebug().withLocation(DUMMY_LOCATION).log(LOG_MESSAGE);
+
+    // then
+    assertMessageMatches("DEBUG");
+  }
+
+  private static Stream<Arguments> severityProvider() {
+    return Stream.of(
+        Arguments.of("DEBUG", Level.TRACE),
+        Arguments.of("DEBUG", Level.DEBUG),
+        Arguments.of("INFO", Level.INFO),
+        Arguments.of("WARNING", Level.WARN));
+  }
+
+  private void assertMessageMatches(final String expectedSeverity) {
+    assertMessageMatchesWithContext(expectedSeverity, null);
+  }
+
+  private void assertMessageMatchesWithContext(
+      final String expectedSeverity, final String expectedContext) {
+    final var message = appender.getEvents().getFirst();
+    final var jsonMessage = encodingAppender.getMessages().getFirst();
+
+    assertThat(jsonMessage)
+        .isEqualTo(
+            """
+        {"timestampSeconds":%d,"timestampNanos":%d,"severity":"%s","message":"Hello World!",\
+        "logging.googleapis.com/sourceLocation":{"file":"Foo.java","line":1,\
+        "function":"Foo.Bar"},%s\
+        "staticContext":{"threadId":%d,"threadName":"%s","threadPriority":%d,\
+        "loggerName":"io.camunda.application.commons.logging.StackdriverLayoutTest"},\
+        "serviceContext":{"service":"","version":""}}""",
+            message.getInstant().getEpochSecond(),
+            message.getInstant().getNanoOfSecond(),
+            expectedSeverity,
+            expectedContext == null ? "" : expectedContext + ",",
+            message.getThreadId(),
+            message.getThreadName(),
+            message.getThreadPriority());
+  }
+
+  private void assertMessageMatchesWithReportLocation() {
+    final var message = appender.getEvents().getFirst();
+    final var jsonMessage = encodingAppender.getMessages().getFirst();
+
+    assertThat(jsonMessage)
+        .isEqualTo(
+            """
```

Comment:
🙃  If you add a `// language=json` comment on it, IntelliJ highlights it nicely and can even auto-format it for you.

Reply:
Yeah, but then it complains that the printf patterns are errors :sweat_smile: